src/core.c/CompUnit/Repository/Installation.pm6

```diff
@@ -23,16 +23,36 @@ class CompUnit::Repository::Installation does CompUnit::Repository::Locally does
         }
     }
 
+    method !prefix-writeable {
+        state $writable = False;
+        once {
```

Comment:
Two things: first I would only cache the result on windows... checking file existence the normal way (by not creating a file) is fine to do repeatedly. Second I would not use `state` or `once` to do this because they are not thread safe and may return undef when a second thread reaches it before the first thread has set the value.

Reply:
`has $!prefix-writable-cache;` + `$!prefix-writable-cache //= do { ... }` is a thread safe way to cache the value (the previously mentioned race condition would instead just write the file again instead of getting back undef)

Reply:
(fwiw similar changes need to be made to zef for its various probes)

Reply:
What's a good alternative? Saving to an attribute and returning that when it's defined?

Reply:
yes, something like `has $!prefix-writable-cache;` + `$!prefix-writable-cache //= do { ... }`